docs/building-on-lisk/web3-app-development.mdx

```diff
@@ -0,0 +1,335 @@
+Synpress can be used with two leading E2E testing frameworks out there, i.e., both [Playwright](https://playwright.dev/) and [Cypress](https://www.cypress.io/), with support for all of their its features.
+
+#### Performance testing
+Performance testing is essential to ensure that your Web3 app can handle the expected load of users and their interactions.
+Measure and improve your apps performance, throughput, and resilience under different levels of traffic and transactions.
+
+Some examples of performance testing tools are:
+
+- [k6](https://k6.io/)
+- [Artillery](https://www.artillery.io/)
+
+{/* #### Automated Testing and Continuous Integration (CI) */}
+
+#### Monitoring
+Monitoring is important to ensure that your Web3 app is running smoothly and to detect and fix issues before they affect your users.
+
+Some popular monitoring tools are:
+
+- [Prometheus](https://prometheus.io/) combined with [Grafana](https://grafana.com/)
+- [Elastic Stack (ELK)](https://www.elastic.co/elastic-stack)
+
+### Web2 vs Web3 App Development
+#### Languages
+Concerning frontend app development, there aren't many differences between Web2 and Web3 apps when it comes to programming languages.
+All languages that are typically used for Web2 app development can also be used for Web3 app development.
+In the end, Web3 apps can even look and feel exactly like Web2 apps for users, because their frontend is built with the same technologies.
+
+However, the backend of Web3 apps is quite different, as it interacts with the blockchain through smart contracts.
+Smart contracts are written in [Solidity](https://soliditylang.org/), therefore Web3 developers need to know Solidity to build the smart contracts they require for their app.
+Luckily, for many use cases, there are already audited smart contracts available(see [OpenZeppelin](https://www.openzeppelin.com/solidity-contracts) or [thirdweb](https://thirdweb.com/explore)) that can be used, so developers don't need to write all of their smart contracts from scratch.
+
+#### Tech Stack
+As mentioned above, the Web2 and Web3 app frontend development is very similar.
+Theoretically, you could build a Web3 app with nearly the same tech stack as a Web2 app, like Next.js, Vite, etc.
+The only thing you need to add is a way for your application to interact with the blockchain, see [Data access](#data-access).
+
+To support developers, many blockchain platforms offer their own SDKs and tools to interact with their blockchain, which can make development easier and more efficient.
+There are also SDKs for Web3 app development that provide a complete tech stack, like [thirdweb](https://thirdweb.com/) or [Alchemy](https://www.alchemy.com/), which take away most of the complexity of blockchain development and provide a smooth developer experience.
+thirdweb supports [Lisk](https://thirdweb.com/lisk) and many other EVM-compatible blockchains.
+
+#### Data Storage
```

Comment:
Perhaps add a few words regarding data privacy and visibility on-chain.
Any data on a public blockchain is also viewable by anyone. Developers need to be aware that:
- Once data is added to a public blockchain, it becomes virtually impossible to remove and is visible to everyone.
- Sensitive or personal data may require additional measures (for example, encryption or zero-knowledge techniques) to avoid unwanted exposure.